### ISSUE TITLE
Unable to install `debugpy` using poetry v1.4.1

### ISSUE THREAD

**Issue by steve-marmalade** [microsoft/debugpy#1246]
Before creating a new issue, please check the [FAQ](https://github.com/microsoft/debugpy/wiki/FAQ) to see if your question is answered there.

## Environment data

- debugpy version: 1.6.6
- OS and version: Arch Linux 6.2.6-arch1-1
- Python version (& distribution if applicable, e.g. Anaconda): 3.11.1
- Using VS Code or Visual Studio: No

## Actual behavior

Running `poetry add debugpy` crashes with a `_WheelFileValidationError`

## Expected  behavior

It installs successfully, just as it does with `pip install debugpy`

## Steps to reproduce:
1. Install Poetry v1.4.1 (I used [the pipx method](https://python-poetry.org/docs/#installing-with-pipx))
2. `mkdir poetry-test && cd poetry-test`
3. `poetry init`
4. `poetry add debugpy`

Given that this issue is present when installing via `poetry` but not `pip`, I initially created the issue in https://github.com/python-poetry/poetry/issues/7686, but the maintainers have directed me to open an issue here to investigate the invalid wheel.

**Comment by twam** [comment]
Same issue here on macOS 13.2.1 with Python 3.11.2.

**Comment by jaidisido** [comment]
Same issue on AWS arm64 docker python3.9 image

**Comment by piiq** [comment]
Working on an m1 macbook.

It seems that poetry 1.4.1 that was released yesterday introduced a couple of regressions. The one mentioned in this issue (_WheelValidationError) and inability to install some packages with a correct md5 wheel hash.

Downgrading poetry to 1.4.0 made the errors go away for me. Hope this helps.

**Comment by DanInacio** [comment]
Had the exact same issue trying to install torch in a Python3.8 environment in a Remote Runner via Github Actions.
Can confirm downgrading Poetry to v1.4.0 solved the problem.

**Comment by hr-curtissimo** [comment]
Confirm that it happens in:
* Windows 11 with poetry 1.4.1
* macOS 12.6.3, 13.2.1 with poetry 1.4.1
* Debian Bullseye with poetry 1.4.1
with all versions of Python 3.9+.

**Comment by gazorby** [comment]
The workaround when using poetry v1.4.1 is to set `installer.modern-installation` setting to false

**Comment by Topchaniuk** [comment]
`installer.modern-installation` - where I can find this setting? Is it in VS Code settings or poetry?

**Comment by yash2tailorbird** [comment]
> `installer.modern-installation` - where I can find this setting? Is it in VS Code settings or poetry?

RUN  `poetry config installer.modern-installation false` in your terminal

**Comment by Topchaniuk** [comment]
@yash2tailorbird  that helped, thank you!

**Comment by ekeric13** [comment]
Is it better practice to set `installer.modern-installation` or to just downgrade to poetry 1.4.1 until the issue is fixed?

**Comment by SH2282000** [comment]
Downgrading to `1.4.0` worked for me.

Here's the command:
```shell
poetry self update 1.4.0
```

**Comment by fredrikaverpil** [comment]
For anyone using poetry in GHA:

```yml
- run: echo "POETRY_VERSION=$(poetry --version)" >> $GITHUB_ENV
  shell: bash

- run: poetry config installer.modern-installation false
  # workaround for bug: https://github.com/microsoft/debugpy/issues/1246
  if: ${{ contains(env.POETRY_VERSION, '1.4.1') }}
  shell: bash
```

**Comment by steve-marmalade** [comment]
While the workarounds are useful to unblock end-users in the short-run, have the maintainers of this repo confirmed whether the `debugpy` wheel is indeed invalid and, if so, scoped the work to resolve it?

**Comment by int19h** [comment]
The files affected are all Win32 binaries that undergo signing in the process of building the official wheels, and need to be re-hashed after that. It seems that this process is broken currently. This is definitely a high-priority issue, and we're investigating.

**Comment by elqurio** [comment]
Works with the new v1.4.2 poetry

**Comment by marmig0404** [comment]
> Works with the new v1.4.2 poetry

But with a warning, per poetry's release notes:

> When trying to install wheels with invalid RECORD files, Poetry does not fail anymore but only prints a warning.

**Comment by int19h** [comment]
Root cause has been identified; a new build of debugpy with fixed hashes is coming shortly.

**Comment by drorata** [comment]
> Works with the new v1.4.2 poetry

For me it doesn't work. I have to configure `poetry config installer.modern-installation false` first. Otherwise, I still get the error:

```
_WheelFileValidationError

  ["In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_cython.cp39-win_amd64.pyd didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/_pydevd_frame_eval/pydevd_frame_evaluator.cp39-win_amd64.pyd didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/attach_amd64.dll didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/attach_x86.dll didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/inject_dll_amd64.exe didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/inject_dll_x86.exe didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/run_code_on_dllmain_amd64.dll didn't match RECORD", "In /Users/fooboo/Library/Caches/pypoetry/artifacts/2e/09/48/001b624046a68d08cdbd434b4da9011c6d44fd1c83f9155679a8edad00/debugpy-1.6.6-py2.py3-none-any.whl, hash / size of debugpy/_vendored/pydevd/pydevd_attach_to_process/run_code_on_dllmain_x86.dll didn't match RECORD"]
```

**Comment by int19h** [comment]
debugpy 1.6.7 is now up on PyPI and should fix this problem. Please let me know if there are any issues.

**Comment by mmabrouk** [comment]
I still get an error:

```
  Output:
  ERROR: debugpy-1.6.7-cp39-cp39-macosx_11_0_x86_64.whl is not a supported wheel on this platform.
  
```

I am using a macos 13.2.1 (22D68)

Installing with pip works by the way. 

I tried installing both with poetry 1.4.0 and 1.4.2, also tried `poetry config installer.modern-installation false` with no avail.

**Comment by mmabrouk** [comment]
I have found a solution to my issue, along with a workaround that may be helpful to others who encounter a similar problem.

Here is the issue as I understand it: the wheel file for debugpy is named `debugpy-1.6.7-cp39-cp39-macosx_11_0_x86_64.whl`, which is not the correct platform for pip (in my case, I am using macOS 13.2.1). When I run `pip install` with the wheel file, I receive the same error as when using Poetry. This is because Poetry is also using pip under the hood to install the wheel file and encountering the same error.

To resolve the issue, follow these steps:

1. Download the wheel file for debugpy from the original source.
2. Rename the wheel file to make it compatible with all platforms by running this command: 
```
mv debugpy-1.6.7-cp39-cp39-none-any.whl debugpy-1.6.7-cp39-none-any.whl
```
3. Remove debugpy from the `poetry.lock` file.
4. Manually add the wheel file to Poetry with the following command:
```
poetry add debugpy-1.6.7-cp39-none-any.whl --platform linux/amd64
```
5. Install as usual with the command:
```
poetry install -E all
```

I hope this helps!

**Comment by mmabrouk** [comment]
For some reason poetry is using `debugpy-1.6.7-cp39-cp39-macosx_11_0_x86_64.whl` and not `none-any`. My solution is to download the `debugpy-1.6.7-cp39-cp39-macosx_11_0_x86_64.whl` and rename it to `none-any` then install it manually. I hope this helps

**Comment by anjanvb** [comment]
> Works with the new v1.4.2 poetry

I can confirm that this is still an issue with v1.4.2 in Intel Macs at least.

**Comment by guidorice** [comment]
poetry 1.7.1 on macos and it looks this is still occurring.  

> ... depends on debugpy (*) which doesn't match any versions, version solving failed.

Has anyone considered that this issue should be re-opened?
